Testscripts/Windows/STRESSTEST-VERIFY-RESTART.ps1

```diff
@@ -17,54 +17,107 @@ function Main {
         [object] $TestProvider
     )
 
-    $result = ""
     $testResult = ""
     $resultArr = @()
+    $keyWord = ""
+    $CurrentTestResult = Create-TestResultObject
 
     foreach ($param in $currentTestData.TestParameters.param) {
         $paramName = $param.Split("=")[0]
         $paramValue = $param.Split("=")[1]
         if ($paramName -eq "rebootNumber") {
             $rebootNumber = $paramValue
         }
+        if ($paramName -eq "type") {
+            $testType = $paramValue
+        }
     }
     if (-not $rebootNumber) {
         $rebootNumber = "1"
     }
-    for ($rebootNr = 1; $rebootNr -le $rebootNumber; $rebootNr++) {
-        try {
-            Write-LogInfo ("Trying to restart {0}: {1} / {2} ..." `
-                -f @($AllVMData.RoleName, $rebootNr, $rebootNumber))
-            $isRestarted = $TestProvider.RestartAllDeployments($allVMData)
-            if ($isRestarted) {
-                Write-LogInfo "Virtual machine restart successful."
-                $testResult = "PASS"
-            } else {
-                Write-LogErr "Virtual machine restart failed."
-                $testResult = "FAIL"
-                break
-            }
-        } catch {
-            $ErrorMessage =  $_.Exception.Message
-            $ErrorLine = $_.InvocationInfo.ScriptLineNumber
-            Write-LogInfo "EXCEPTION : $ErrorMessage at line: $ErrorLine"
+    $expectedcount = 0
+    switch ($testType) {
+        {($_ -imatch "NVME")} {
+            $keyWord = "Non-Volatile memory controller";
+            $expectedcount = Get-ExpectedCount -ip $AllVmData.PublicIP -port $AllVmData.SSHPort -username $user -password $password -size $AllVmData.InstanceSize -type "NVME"
```

Comment:
We can use just one line of '$output= Get-ExpectedCount -ip .....',  then split $output into $expectedCount and $keyWord.  Yes, modify the common function Get-ExpectedCount
Remember we should follow a proper naming rule, the local variable should be 'camelCase', so $expectedCount not $expectedcount, just like $keyWord